### PR TITLE
Improve scheduling queue's logic

### PR DESCRIPTION
**Pull request by bsalamat**
**What this PR does / why we need it**:
Improves scheduling queue's code based on some recent comments on [the original PR](https://github.com/kubernetes/kubernetes/pull/55109).
This PR does not fix any bugs or make any change of behavior.

**Release note**:

```release-note
NONE
```

/sig scheduling

### CODE REVIEW

File: pkg/scheduler/core/scheduling_queue.go [hunk 1]
```diff
@@ -410,11 +410,9 @@ func (p *PriorityQueue) AssignedPodUpdated(pod *v1.Pod) {
 func (p *PriorityQueue) MoveAllToActiveQueue() {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	var unschedulablePods []interface{}
 	for _, pod := range p.unschedulableQ.pods {
-		unschedulablePods = append(unschedulablePods, pod)
+		p.activeQ.Add(pod)
```
Comment (k82cn) [inline pkg/scheduler/core/scheduling_queue.go]: Should we log error message if return error?

File: pkg/scheduler/core/scheduling_queue.go [hunk 2]
```diff
@@ -424,8 +422,9 @@ func (p *PriorityQueue) movePodsToActiveQueue(pods []*v1.Pod) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	for _, pod := range pods {
-		p.activeQ.Add(pod)
-		p.unschedulableQ.Delete(pod)
+		if p.activeQ.Add(pod) == nil {
```
Comment (k82cn) [inline pkg/scheduler/core/scheduling_queue.go]: ditto

**Review by k82cn — APPROVED** [review]
LGTM, just tiny comments :)

File: pkg/scheduler/core/scheduling_queue.go (hunk 1 above)
Comment (bsalamat) [reply pkg/scheduler/core/scheduling_queue.go]: The `Add()` function logs the error. We don't need to log again here.

Comment (k82cn) [reply pkg/scheduler/core/scheduling_queue.go]: `Heap.Add` just return error, `PriorityQueue.Add` logged error; `unschedulableQ` and `activeQ` are using different keyFunc :)

**Comment by bsalamat** [comment]
Thanks, @k82cn!

**Pull request merged**